model/generate.go

```diff
@@ -270,10 +270,16 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, v *Version
 		return errors.Errorf("project '%s' not found", p.Identifier)
 	}
 
+	allTasksToBeCreatedIncludingDeps, err := NewPatchTaskIdTable(p, v, *newTVPairs, projectRef.Identifier)
```

Comment:
If `allTasksToBeCreatedIncludingDeps` is created from `newTVPairs`, which is the global set of tasks to create, and we're passing either `newTVPairsForExistingVariants` or `newTVPairsForNewVariants` (depending whether we're calling `addNewTasks` or `addNewBuilds`) to `getTaskIdTables` [here](https://github.com/evergreen-ci/evergreen/pull/7149/files#diff-f50207584681d142f6e119549a8e2e22249b45dcbe3cb6f7570eb906be64299dR1803), then wouldn't the merging logic [here](https://github.com/evergreen-ci/evergreen/pull/7149/files#diff-f50207584681d142f6e119549a8e2e22249b45dcbe3cb6f7570eb906be64299dR1819-R1841) just no-op?

From my reading at least, it looks like any new table we calculate in `getTaskIdTables` would already be existing in the larger map that we're passing in [here](https://github.com/evergreen-ci/evergreen/pull/7149/files#diff-96655072263d498b6dc1571c45fc69e886abcc4bc9994343a80af2f54375c7bcR282)

Reply:
Hmmm, I see. Looks like `addNewTasks` [does it too](https://gist.github.com/hadjri/591b868070b8e73aaf2355fb385188c4#file-lifecycle-go-L1708) IIUC. That makes sense, thanks for the explanation and for grappling with the general complexity here